src/sched/lthread.c

```diff
@@ -474,23 +476,32 @@ int __init_utp(void *p, int set_tp)
 void *__copy_utls(struct lthread *lt, unsigned char *mem, size_t sz)
 {
 	mem += sz - sizeof(struct lthread_tcb_base);
-	mem -= (uintptr_t)mem & (libc.tls_align - 1);
+	mem -= (uintptr_t)mem & (TLS_ALIGN - 1);
 	return (void *)mem;
 }
 
 void set_tls_tp(struct lthread* lt)
 {
+    uintptr_t tp_unaligned, tp;
     if (!lt->itls)
         return;
-
-    uintptr_t tp_unaligned =
+    // pthread_create passes thread pointer for lt->itls
+    // no need to calculate offset of tp in tls region
+    if (lt->attr.stack_size == 0)
```

Comment:
Sorry, I may have just broken this logic with the deferred stack freeing PR.  It's probably worth adding a flag in the lthread structure to check if this is a cloned task or not.  That said, I'd prefer if we always just set `tp = lt->itls` unconditionally and modified the other code path that set itls to do the adjustment.

Reply:
Adding thread_type to lthread_attr